lua/ttt2/libraries/marker_vision.lua

```diff
@@ -21,160 +22,203 @@ markerVision = {}
--- @param Entity ent The entity that should be rendered
--- @param Player owner The owner of the wallhack that takes their ownershipo with them on team change
--- @param number visibleFor Visibility setting: `VISIBLE_FOR_PLAYER`, `VISIBLE_FOR_TEAM`, `VISIBLE_FOR_ALL`
--- @param[opt] Color color The color of the wallhack, uses team color as fallback
--- @param[opt] table receiverList A list of players that should receive the netmessage, overwrites the default
--- @param[opt] any passThroughData any data that should be added to this radar vision, it is also synced to the client
--- @note Call on server to add entity on server and all defined clients.
+-- Creates a new marker vision object for the entity.
+-- @note This does not sync to the client, @{MARKER_VISION_OBJECT:SyncToClients}
+-- has to be called on it first
+-- @param Entity ent The entity which should receive the marker vision element
+-- @param string identifier The unique identifier of this marker vision element
+-- @return MARKER_VISION_OBJECT The marker vision object that was created
 -- @realm shared
-function markerVision.RegisterEntity(ent, owner, visibleFor, color, receiverList, passThroughData)
-	if SERVER then
-		local plysTeam = GetTeamFilter(owner:GetTeam(), true, true)
-
-		if #plysTeam == 0 then -- happens for TEAM_NONE for example
-			visibleFor = VISIBLE_FOR_PLAYER
-		end
+function markerVision.Add(ent, identifier)
+	local mvObject = table.Copy(MARKER_VISION_OBJECT)
+	mvObject:SetEnt(ent)
+	mvObject:SetIdentifier(identifier)
 
-		local streamTable = {
-			ent = ent,
-			owner = owner,
-			visibleFor = visibleFor,
-			color = color,
-			data = passThroughData
-		}
+	markerVision.registry[#markerVision.registry + 1] = mvObject
 
-		local receiver
-
-		if visibleFor == VISIBLE_FOR_PLAYER then
-			receiver = owner
-		elseif visibleFor == VISIBLE_FOR_TEAM then
-			receiver = plysTeam
-		end
-
-		-- note: when VISIBLE_FOR_ALL is used, the receiver will be nil and therefore SendStream uses net.Broadcast
-		net.SendStream("ttt2_register_entity_added", streamTable, receiverList or receiver)
-	end
-
-	if CLIENT then
-		if visibleFor == VISIBLE_FOR_ALL then
-			color = color or TEAMS[TEAM_INNOCENT].color
-		elseif visibleFor == VISIBLE_FOR_PLAYER then
-			color = color or TEAMS[TEAM_NONE].color
-		else
-			color = color or TEAMS[LocalPlayer():GetTeam()].color
-		end
-
-		marks.Add({ent}, color)
-	end
-
-	markerVision.registry[ent] = {
-		owner = owner,
-		visibleFor = visibleFor,
-		color = color,
-		data = passThroughData
-	}
+	return mvObject
 end
 
 ---
--- Returns the visibleFor flag for a provided entity, nil if not set
--- @param Entity ent The tracked entity
--- @return number|nil The visibility flag
+-- Returns the marker vision element if it exists.
+-- @param Entity ent The entity which should receive the marker vision element
+-- @param string identifier The unique identifier of this marker vision element
+-- @return MARKER_VISION_OBJECT The marker vision object
```

Comment:
Missing the 2nd return value.
And missing the possible nil case

Also is the second param even needed?

Reply:
Yew it is needed as only a combination of ent and identifier is unique

Reply:
I meant second return value

Reply:
This is also needed. Otherwise I'd need a direct table access eith the index which I try to avoid

Reply:
It is still undocumented. Thats all what this is about.

Reply:
oh, sorry, I misunderstood this